datadotworld/client/api.py

```diff
@@ -0,0 +1,451 @@
+            self._datasets_api.replace_dataset(owner_id, dataset_id, request)
+        except _swagger.ApiException as e:
+            raise RestApiError(cause=e)
+
+    # File Operations
+
+    def add_files_via_url(self, dataset_key, files={}):
+        """Add or update dataset files linked to source URLs
+
+        Parameters
+        ----------
+        dataset_key : str
+            Dataset identifier, in the form of owner/id
+        files : dict
+            File names and source URLs to add or update
+
+        Raises
+        ------
+        RestApiException
+            If a server error occurs
+
+        Examples
+        --------
+        >>> api_client = datadotworld.api_client()
+        >>> api_client.add_files_via_url('jonloyens/an-intro-to-dataworld-dataset',  
+        >>>    {'atx_startup_league_ranking.csv': 'http://www.atxsa.com/sports/basketball/startup_league_ranking.csv'})
+        """
+        file_requests = [_swagger.FileCreateOrUpdateRequest(
+            name=name, source=_swagger.FileSourceCreateOrUpdateRequest(url=url)) for name, url in files.items()]
+
+        owner_id, dataset_id = parse_dataset_key(dataset_key)
+        try:
+            self._datasets_api.add_files_by_source(owner_id, dataset_id,
+                                                   _swagger.FileBatchUpdateRequest(files=file_requests))
+        except _swagger.ApiException as e:
+            raise RestApiError(cause=e)
+
+    def sync_files(self, dataset_key):
+        """Trigger synchronization process to update all dataset files linked to source URLs
+
+        Parameters
+        ----------
+        dataset_key : str
+            Dataset identifier, in the form of owner/id
+
+        Raises
+        ------
+        RestApiException
+            If a server error occurs
+
+        Examples
+        --------
+        >>> api_client = datadotworld.api_client()
+        >>> api_client.sync_files('jonloyens/an-intro-to-dataworld-dataset')
+        """
+        try:
+            self._datasets_api.sync(*(parse_dataset_key(dataset_key)))
+        except _swagger.ApiException as e:
+            raise RestApiError(cause=e)
+
+    def upload_files(self, dataset_key, files):
+        """Upload dataset files
+
+        Parameters
+        ----------
+        dataset_key : str
+            Dataset identifier, in the form of owner/id
+        files : list of str
+            The list of names/paths for files stored in the local filesystem
+
+        Raises
+        ------
+        RestApiException
+            If a server error occurs
+
+        Examples
+        --------
+        >>> api_client = datadotworld.api_client()
+        >>> api_client.upload_files('jonloyens/an-intro-to-dataworld-dataset',
+        >>>                 ['/Users/jon/DataDotWorldBBall/DataDotWorldBBallTeam.csv'])
+        """
+        owner_id, dataset_id = parse_dataset_key(dataset_key)
+        try:
+            self._uploads_api.upload_files(owner_id, dataset_id, files)
```

Comment:
This passes a list of files but the uploads_api signature accepts a single file.